data-structures.md

```diff
@@ -29,22 +29,47 @@ For most objects referenced by Filecoin, a Content Identifier (CID for short) is
 
 CIDs are serialized by applying binary multibase encoding, then encoding that as a CBOR byte array with a tag of 42.
 
+## Tickets
+
+A ticket is Fielcoin's version of a cryptographic lottery ticket and is used to prove that one is a block leader. At a high-level, tickets must do the following:
+
+- Ensure leader secrecy -- meaning a block producer will not be known until they release their block to the network.
+- Prove leader election -- meaning a block producer can be verified by any participant in the network.
+- Prove appropriate delay between drawings — thereby preventing leaders from "rushing" the protocol by releasing blocks early (at the expense of fairness for miners with worse connectivity).
+- Ensure a single drawing per round — derived in part from the above, thereby preventing miners from grinding on tickets (e.g. by repeatedly drawing new tickets in the hopes of winning) within a round.
+
+```go
+type Ticket struct {
+  
+	// The VDF Result is derived from the prior ticket in the ticket chain
+	VDFResult BigInteger
```

Comment:
@sternhenri, @ZenGround0: Do we have a language-independent description of BigInteger here? What is the binary representation? Maybe we could point to a spec that defines this for us.

Reply:
That is what LEB128 is for

Reply:
Do we have a size for these bigInts?

Reply:
They are variably length sized, to support arbitrary numbers. But if you have a max range you want to represent then I can tell you the size.

Reply:
Understood. Someone will have to update the spec comments to expound on this. As for the typing, we are using inconsistent type nomenclature for big integers. It would be nice to go through and standardize these. So far I've seen `Integer`, `BigInteger`, `BigInt`; they no doubt mean the same thing --> LEB128 encoded integers of arbitrary size but we should specifically call that out here. 

If the spec owners acknowledge that this is what they mean when they write an element from the list above, I'll go ahead and change it. Don't want to misinterpret the authors intent.